### DISCUSSION TITLE
Bad register code (65536).

### DISCUSSION THREAD

**Discussion by pddzaic** [lvjr/tabularray#149] [Q&A]
Before writing an bug issue i would like to understand, what the error message means.

I have a huge table: 10 columns, 1000 rows. The compiling process (lualatex) stops with the following error message:

```
[! Bad register code (65536).
\__regex_query_set_aux:nN ...__regex_curr_pos_int
                                                  {#1}\__kernel_intarray_gse...

l.842 \end
        {longtblr}
```

This happens after 500 rows. 

Here the minimal working example:

```
\documentclass{standalone}

\usepackage{tabularray}
\newcommand\id[1]{\lstinline{#1}}

\begin{document}
\input{test.tex}
\end{document}
```

I have attached the `test.tex`.

[test.zip](https://github.com/lvjr/tabularray/files/7497026/test.zip)

**Comment by lvjr** [comment]
I guess it is due to the limitations of l3regex and l3intarray of LaTeX3 kernel. Please minimize your TeX file and share it.

**Reply by pddzaic** [reply]
The file is attached.

**Reply by josephwright** [reply]
I agree this is going to be an `l3regex` issue: the question is why this doesn't get trapped properly.

**Reply by pddzaic** [reply]
Where should I post a bug report?

**Reply by PhelypeOleinik** [reply]
@pddzaic There's not really a bug to fix.  `l3regex` uses toks registers to do its job, but there are only 65535 of those available in LuaTeX (half that amount in other engines) so you are facing a limitation of the engine.

On the `l3regex` side the best we can reasonably do is to check if that limit is reached and throw a better error message.

It's not a bug in `tabularray` either (it does work for not-so-huge tables), but maybe it could replace its uses of regex by simpler alternatives, or chop the table in smaller chunks (somehow) and apply the regex on each part (no idea how it works internally, so what I said may be complete garbage).

**Reply by lvjr** [reply]
Maybe `l3regex` could use token lists instead of toks registers to do its job.

**Reply by josephwright** [reply]
@lvjr No, the trick is that `\toks` can be referred to by number: this allows very fast usage of a large set of storage bins.

**Comment by lvjr** [comment]
This bug has been fixed. See issue #305.
